### PR TITLE
close #2509 reset with empty argument

### LINKED ISSUE #2509: Different behavior of register and register() after reset()
**Describe the bug**
After calling reset(), fields that have register() and register behave differently.

**To Reproduce**
1. Create fields that have register() and register.
```
        <input name="field1" ref={register()} />
        <input name="field2" ref={register} />
```
2. Call reset()
```
  const onReset = useCallback((e) => {
    e.preventDefault();
    reset();
  }, [reset]);
```

3. Call getValues(). It returns only a value of field1.
```
  const onDump = useCallback((e) => {
    e.preventDefault();
    console.log("values");
    console.log(getValues());
  }, [getValues]);
```

**Codesandbox link (Required)**
https://codesandbox.io/s/zen-buck-ujt63?file=/src/App.js

**Expected behavior**
getValues() returns values of field1 and field2.

**Desktop (please complete the following information):**
- OS: Ubuntu 20.04
- Browser: Firefox
- Version: 78

### PR DESCRIPTION
**Pull request by bluebill1049**
https://codesandbox.io/s/heuristic-feistel-qxl8h?file=/src/App.js

### CODE REVIEW

File: src/controller.test.tsx [hunk 1]
```diff
@@ -81,7 +81,9 @@ describe('Controller', () => {
             as={<input />}
             control={control}
           />
-          <button onClick={() => reset()}>reset</button>
+          <button type="button" onClick={() => reset()}>
```
Comment (bluebill1049) [inline src/controller.test.tsx:84]: @keiya01 could you take a look at this test? I couldn't figure out why it's failing, tested in CSB and it's working :(

Comment (keiya01) [reply src/controller.test.tsx:84]: I will check tomorrow 👀

Comment (bluebill1049) [reply src/controller.test.tsx:84]: thanks keiya. here is a test: https://codesandbox.io/s/heuristic-feistel-qxl8h?file=/src/App.js in csb

Comment (keiya01) [reply src/controller.test.tsx:84]: Probably this CSB is importing module from `react-hook-form` but we should import module from './src'.🤔

Comment (keiya01) [reply src/controller.test.tsx:84]: And if I change to import from `./src`, `reset` don't work correctly.

Comment (bluebill1049) [reply src/controller.test.tsx:84]: oh oops, stupid of me... will investigate for a fix.

Comment (keiya01) [reply src/controller.test.tsx:84]: I found a wrong code. Could I push the change ?

Comment (bluebill1049) [reply src/controller.test.tsx:84]: YES! PLZ

**Comment by keiya01** [comment]
@bluebill1049 I have completed the change.  Could you review ?

**Comment by bluebill1049** [comment]
oh wow! thank you so much for solving this issue @keiya01 🚀

File: src/controller.tsx [hunk 2]
```diff
@@ -119,6 +119,9 @@ const Controller = <
         }),
         rules,
       );
+      if (isNotFieldArray && !get(defaultValuesRef.current, name)) {
```
Comment (bluebill1049) [inline src/controller.tsx:122]: tricky one. 🏄🏻‍♂️

**Review by bluebill1049 — COMMENTED** [review]
Awesome work! tested too. it's working!

File: src/controller.server.test.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { Controller } from './controller';
+import { useForm } from './useForm';
+import { renderToString } from 'react-dom/server';
+
+describe('Controller with SSR', () => {
+  /**
+   * This test is checking https://github.com/react-hook-form/react-hook-form/issues/1398
+   */
+  it('should render correctly with as with component', () => {
```
Comment (bluebill1049) [inline src/controller.server.test.tsx:10]: oh! thanks for this.

**Pull request merged**